### DISCUSSION TITLE
[Urgent] How can I recover a deleted REALM

### DISCUSSION THREAD

**Discussion by ghost** [keycloak/keycloak#15696] [Q&A]
Hi,
I just deleted a realm manually thinking it is a user that I'm deleting. What are different options to recover the Realm ?
To my understanding, in DB a realm deleted would not delete other resources right ? What about recreating one with the same name ? 
Anyway, I'm working with my team, new to keycloak, so I'm gathering information for them, I know very little about keycloak and roles.
My stress is skyhigh, as it escalated to production and everything is down

**Comment by sschu** [comment]
Deleting a realm does delete most associated data. Do you have a database backup?

**Reply by ghost** [reply]
It looks like they had some virtualization at some point. probably docker or a VM or a physical disk, I don't know. It was resolved "somehow".
True, they asked for some users roles which we had thankfully in another database.
This is strange though, isn't it wiser not to delete every relation when a realm is deleted ?

I learnt something anyway and problem was solved after headaches , thanks !

**Reply by ghost** [reply]
also if you ask me how I could mistaken deleting a Realm from a User, true, it's a shame,
but if you check keycloak web UI, you have definitely the same button same design for two listings. I suggest to add some safety like GitHub does for some actions: _Enter the name of the realm to confirm_ or so

**Reply by sschu** [reply]
The thing is basically all entities are sub-resources of realms and are accessed using URLs including the realm, so changing the deletion behaviour is pretty much impossible. The safety check for deleting realms could be a bit more involved, I agree.
